### ISSUE TITLE
Classvar[Optional[Tuple]] must be unpacked

### ISSUE THREAD

**Issue by KKawamura1** [PyCQA/pylint#2855]
Pylint fails to recognize that the class variable is not None.

Code: 

```python
from typing import ClassVar, Optional, Tuple


class A:
    pass


class B:
    pass


A_B = Tuple[A, B]


class C:
    a_b: ClassVar[Optional[A_B]] = None

    @classmethod
    def cached_a_and_b(cls) -> A_B:
        if cls.a_b is None:
            cls.a_b = (A(), B())
        return cls.a_b

    @classmethod
    def f(cls) -> None:
        a, b = cls.cached_a_and_b()  # Must be unpacked
        print(a, b)
```

`pylint <name_of_above_file>.py` produces

```
************* Module test
...

test.py:26:8: E0633: Attempting to unpack a non-sequence defined at line 16 (unpacking-non-sequence)

...
```

which is a fail because the `cls.cached_a_and_b()` must return `Tuple[A, B]`.

### Steps to reproduce
1. Write above scripts.
2. Run `pylint <path-to-above-script>`.
3. That's all.

### Current behavior

At line 26 (`a, b = cls.cached_a_and_b()`), pylint argues that there is `E0633`.

### Expected behavior

No error would be emmited.

### pylint --version output

pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Mar 27 2019, 09:23:15)
[Clang 10.0.1 (clang-1001.0.46.3)]

**Comment by PCManticore** [comment]
Thanks for the report. As of now, `pylint` does not have support for type annotations, it does not consider them when inferring the type of the objects involved in analysis. In this case though, it might infer the value of `cached_a_and_b` as `None`, leading to the error you see.
